x-pack/metricbeat/module/sql/query/query.go

```diff
@@ -33,94 +32,142 @@ func init() {
 	)
 }
 
+type query struct {
+	Query          string `config:"query" validate:"nonzero,required"`
+	ResponseFormat string `config:"response_format" validate:"nonzero,required"`
+}
+
 // MetricSet holds any configuration or state information. It must implement
 // the mb.MetricSet interface. And this is best achieved by embedding
 // mb.BaseMetricSet because it implements all of the required mb.MetricSet
 // interface methods except for Fetch.
 type MetricSet struct {
 	mb.BaseMetricSet
-	Driver         string
-	Query          string
-	ResponseFormat string
-
+	Config struct {
+		Driver         string `config:"driver" validate:"nonzero,required"`
+		ResponseFormat string `config:"sql_response_format"`
+		// New flag
+		RawData rawData `config:"raw_data"`
+		// Support either the previous query / or the new list of queries.
+		Query   string  `config:"sql_query" `
+		Queries []query `config:"sql_queries" `
+	}
 	db *sqlx.DB
 }
 
+// rawData is the minimum required set of fields to generate fully customized events with their own module key space
+// and their own metricset key space.
+type rawData struct {
+	Enabled bool `config:"enabled"`
+}
+
 // New creates a new instance of the MetricSet. New is responsible for unpacking
 // any MetricSet specific configuration options if there are any.
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	cfgwarn.Beta("The sql query metricset is beta.")
 
-	config := struct {
-		Driver         string `config:"driver" validate:"nonzero,required"`
-		Query          string `config:"sql_query" validate:"nonzero,required"`
-		ResponseFormat string `config:"sql_response_format"`
-	}{ResponseFormat: tableResponseFormat}
+	b := &MetricSet{BaseMetricSet: base}
 
-	if err := base.Module().UnpackConfig(&config); err != nil {
+	if err := base.Module().UnpackConfig(&b.Config); err != nil {
 		return nil, err
 	}
 
-	if config.ResponseFormat != variableResponseFormat && config.ResponseFormat != tableResponseFormat {
-		return nil, fmt.Errorf("invalid sql_response_format value: %s", config.ResponseFormat)
+	if b.Config.ResponseFormat != "" && b.Config.ResponseFormat != variableResponseFormat && b.Config.ResponseFormat != tableResponseFormat {
```

Comment:
What happens if `b.Config.ResponseFormat == ""`?

Reply:
Fixed to cover this scenario.